src/test/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScannerTest.java

```diff
@@ -406,6 +404,74 @@ public void testLeastCommonAncestor() throws Exception {
+                        "        },\n" +
+                        "        'otherunit': {\n" +
+                        "            retry(1) {\n" +
+                        "              sleep 1;\n" +
+                        "              sleep 5; \n" +
+                        "              echo 'goodbye'   \n" +
+                        "            }\n" +
+                        "        }\n" +  // end of branch:
+                        "    ])\n"
+        ));
+        /*Node dump follows, format:
+        [ID]{parent,ids}(millisSinceStartOfRun) flowNodeClassName stepDisplayName [st=startId if a block end node]
+        Action format:
+        - actionClassName actionDisplayName
+        ------------------------------------------------------------------------------------------
+        [2]{}FlowStartNode Start of Pipeline
+        [3]{2}StepAtomNode test
+        [4]{3}StepStartNode Execute in parallel : Start
+        [6]{4}StepStartNode Branch: unit
+        [7]{4}StepStartNode Branch: otherunit
+            A [8]{6}StepStartNode Retry the body up to N times : Start
+            A [9]{8}StepStartNode Retry the body up to N times : Body : Start
+          B [10]{7}StepStartNode Retry the body up to N times : Start
+          B [11]{10}StepStartNode Retry the body up to N times : Body : Start
+            A [12]{9}StepAtomNode Sleep
+          B [13]{11}StepAtomNode Sleep
+            A [14]{12}StepAtomNode Sleep
+          B [15]{13}StepAtomNode Sleep
+          B [16]{15}StepAtomNode Print Message
+          B [17]{16}StepEndNode Retry the body up to N times : Body : End  [st=11]
+          B [18]{17}StepEndNode Retry the body up to N times : End  [st=10]
+          B [19]{18}StepEndNode Execute in parallel : Body : End  [st=7]
+            A [20]{14}StepAtomNode Print Message
+            A [21]{20}StepEndNode Retry the body up to N times : Body : End  [st=9]
+            A [22]{21}StepEndNode Retry the body up to N times : End  [st=8]
+            A [23]{22}StepEndNode Execute in parallel : Body : End  [st=6]
+        [24]{23,19}StepEndNode Execute in parallel : End  [st=4]
+        [25]{24}FlowEndNode End of Pipeline  [st=2]*/
+        WorkflowRun b = r.assertBuildStatusSuccess(job.scheduleBuild2(0));
+        FlowExecution exec = b.getExecution();
+        ForkScanner scan = new ForkScanner();
+
+        // Test different start points in branch A & B, 20 and 19 were one error case.
+        for (int i=0; i < 4; i++) {
+            for (int j=0; j<5; j++) {
+                int branchANodeId = i+20;
+                int branchBNodeId = j+15;
+                System.out.println("Starting test with nodes "+branchANodeId+","+branchBNodeId);
+                ArrayList<FlowNode> starts = new ArrayList<FlowNode>();
+                FlowTestUtils.addNodesById(starts, exec, branchANodeId, branchBNodeId);
+                List<FlowNode> all = scan.filteredNodes(starts, Predicates.<FlowNode>alwaysTrue());
```

Comment:
Nope -- it's an iterator so there are a bunch of trivial one-liners that will accomplish that goal.  This is one but there are a ton you can do using Guava.